usr/src/uts/common/fs/zfs/sys/spa.h

```diff
@@ -409,6 +489,26 @@ _NOTE(CONSTCOND) } while (0)
 #define	BP_GET_LEVEL(bp)		BF64_GET((bp)->blk_prop, 56, 5)
 #define	BP_SET_LEVEL(bp, x)		BF64_SET((bp)->blk_prop, 56, 5, x)
 
+/* encrypted, authenticated, and MAC cksum bps use the same bit */
+#define	BP_USES_CRYPT(bp)		BF64_GET((bp)->blk_prop, 61, 1)
+#define	BP_SET_CRYPT(bp, x)		BF64_SET((bp)->blk_prop, 61, 1, x)
+
+#define	BP_IS_ENCRYPTED(bp)			\
+	(BP_USES_CRYPT(bp) &&			\
+	BP_GET_LEVEL(bp) == 0 &&		\
```

Comment:
The ZoL version has `<= 0`.  That said, I think that BP_GET_LEVEL() will result in an unsigned value, so it can never be <0.  And the BP_GET_LEVEL() is never negative-ish (`ZB_ZIL_LEVEL` and `ZB_ROOT_LEVEL` are not used here, those block types have BP_GET_LEVEL()==0).  So I think we should change ZoL to use `==` here, @tcaputi.